pkg/webhook/validating/pod.go

```diff
@@ -72,7 +72,34 @@ func PodCreateValidation(ctx context.Context, req *admission.Request, handler *H
 		} else {
 			networkType = ipamtypes.ParseNetworkTypeFromString(string(networkingv1.GetNetworkType(network)))
 		}
+	}
+
+	// Specified Subnet Validation
+	var specifiedSubnet string
+	if specifiedSubnet = utils.PickFirstNonEmptyString(pod.Annotations[constants.AnnotationSpecifiedSubnet], pod.Labels[constants.LabelSpecifiedSubnet]); len(specifiedSubnet) > 0 {
+		subnet := &networkingv1.Subnet{}
+		if err = handler.Cache.Get(ctx, types.NamespacedName{Name: specifiedSubnet}, subnet); err != nil {
+			if errors.IsNotFound(err) {
+				return admission.Denied(fmt.Sprintf("specified subnet %s not found", specifiedSubnet))
+			}
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+
+		if len(specifiedNetwork) > 0 {
+			if subnet.Spec.Network != specifiedNetwork {
+				return admission.Denied(fmt.Sprintf("specified subnet %s not belong to specified network %s",
+					specifiedSubnet,
+					specifiedNetwork,
+				))
+			}
+		} else {
+			// subnet can also determine the specified network
+			specifiedNetwork = subnet.Spec.Network
+		}
+	}
 
+	// Existing IP Instances Validation
+	if len(specifiedNetwork) > 0 {
```

Comment:
Do we need to check network type again here? If there is specified subnet only.

Reply:
network not network-type here

Reply:
If a subnet is specified, then the network will be determined by this. Of course, we should check potential conflicts of network and existing IP instances.

Reply:
> network not network-type here

I mean, this line is fine. But if the network and network type of pod are not specified manually, `networkType`  will be empty here even if we have a specified subnet. Maybe `networkType` should be decided again above this line.